test/integration/cni-egress/pod_egress_suite_test.go

```diff
@@ -0,0 +1,138 @@
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "CNI Pod Networking Suite")
+}
+
+var _ = BeforeSuite(func() {
+	f = framework.New(framework.GlobalOptions)
+
+	By("checking cluster v4 or v6")
+	clusterOutput, err := f.CloudServices.EKS().DescribeCluster(f.Options.ClusterName)
+	Expect(err).NotTo(HaveOccurred())
+	isIPv4Cluster = false
+	if *clusterOutput.Cluster.KubernetesNetworkConfig.IpFamily == "ipv4" {
+		isIPv4Cluster = true
+	}
+	By("creating test namespace")
+	f.K8sResourceManagers.NamespaceManager().
+		CreateNamespace(utils.DefaultTestNamespace)
+
+	By(fmt.Sprintf("getting the node with the node label key %s and value %s",
+		f.Options.NgNameLabelKey, f.Options.NgNameLabelVal))
+	nodes, err := f.K8sResourceManagers.NodeManager().GetNodes(f.Options.NgNameLabelKey, f.Options.NgNameLabelVal)
+	Expect(err).ToNot(HaveOccurred())
+
+	By("verifying at least 1 node present for the test")
+	Expect(len(nodes.Items)).Should(BeNumerically(">", 0))
+
+	// Set the primary and secondary node for testing, these are used for a pod traffic test between two pods
+	for _, node := range nodes.Items {
+		if len(node.Spec.Taints) == 0 {
+			if primaryNode.Name == "" {
+				primaryNode = node
+				break
+			}
+		}
+	}
+	Expect(primaryNode.Name).To(Not(HaveLen(0)), "expected to find a non-tainted node")
+
+	// Get the node security group
+	instanceID := k8sUtils.GetInstanceIDFromNode(primaryNode)
+	primaryInstance, err := f.CloudServices.EC2().DescribeInstance(instanceID)
+	Expect(err).ToNot(HaveOccurred())
+
+	if isIPv4Cluster {
+		primaryNodeIp = primaryInstance.Ipv6Address
+	} else {
+		primaryNodeIp = primaryInstance.PublicIpAddress
+	}
+	Expect(primaryNodeIp).NotTo(BeNil())
+
+	By("getting the instance type from node label " + InstanceTypeNodeLabelKey)
+	instanceType := primaryNode.Labels[InstanceTypeNodeLabelKey]
+
+	By("getting the network interface details from ec2")
+	instanceOutput, err := f.CloudServices.EC2().DescribeInstanceType(instanceType)
+	Expect(err).ToNot(HaveOccurred())
+
+	// Subtract 2 for coredns pods if any, both could be on same Interface
+	maxIPPerInterface = int(*instanceOutput[0].NetworkInfo.Ipv4AddressesPerInterface) - 2
+
+	if isIPv4Cluster {
+		k8sUtils.AddEnvVarToDaemonSetAndWaitTillUpdated(f, "aws-node", "kube-system",
+			"aws-node", map[string]string{
+				"ENABLE_V6_EGRESS": "true",
+			})
+		k8sUtils.AddEnvVarToDaemonSetAndWaitTillUpdated(f, "aws-node", "kube-system",
+			"aws-vpc-cni-init", map[string]string{
+				"ENABLE_V6_EGRESS": "true",
+			})
+	}
+})
+
+var _ = AfterSuite(func() {
+	By("deleting test namespace")
+	f.K8sResourceManagers.NamespaceManager().
+		DeleteAndWaitTillNamespaceDeleted(utils.DefaultTestNamespace)
+
+	By("reset daemonset back")
+	k8sUtils.RemoveVarFromDaemonSetAndWaitTillUpdated(f, "aws-node", "kube-system",
+		"aws-vpc-cni-init", map[string]struct{}{
+			"ENABLE_V6_EGRESS": {},
+		})
+	k8sUtils.UpdateEnvVarOnDaemonSetAndWaitUntilReady(f, "aws-node", "kube-system",
```

Comment:
Only `ENABLE_V6_EGRESS` needs to be touched by this function, as the others were not modified in this suite